### ISSUE TITLE
`lc.remove_outliers()` raised "ValueError: could not broadcast input array from shape ..." in some environments

### ISSUE THREAD

**Issue by Nschanche** [lightkurve/lightkurve#1263]
lc.estimate_cdpp fails with the error "ValueError: could not broadcast input array from shape (16781,) into shape ()". The problem is in the call to SigmaClip and then remove_outliers. A quick fix is to modify the remove_outliers call to sigma_clip by changing the data to self.flux.value (rather than self.flux)

**Comment by orionlee** [comment]
I cannot reproduce the problem. Please include specific examples, preferably a reproducible snippet that illustrates the problem.

E.g., in the following snippet, it produces `34.706248ppm` for pi Men in TESS sector 1.

```
import lightkurve as lk

lc = lk.search_lightcurve("pi Men").download()
lc.estimate_cdpp()
```

**Comment by cdevaneprugh** [comment]
I have experienced the same problem, specifically with premade light curves. It looks like the issue pops up when you try to use `.remove_outliers()` after using `.flatten()`. For example, if I try: 

```
import lightkurve as lk

lc = lk.search_lightcurve('KIC 6922244', author="Kepler", 
                          cadence="long", quarter=4).download()
lc = lc.flatten()
lc = lc.remove_outliers()
```

I get a value error that says:  `could not broadcast input array from shape (4116,) into shape ()`

However, if I download a tpf file, then convert it to a light curve (like below), there is no issue.
```
tpf = lk.search_targetpixelfile('KIC 6922244', author="Kepler", 
                                cadence="long", quarter=4).download()
lc = tpf.to_lightcurve(aperture_mask=tpf.pipeline_mask)
lc = lc.flatten()
lc = lc.remove_outliers()
```

I also fixed the issue in a similar way, by removing the astropy units from the flux array in `.remove_outliers()`. It looks like the issue popped up when Astropy updated to version 5.0

**Comment by orionlee** [comment]
Hmm, I cannot reproduce the problem indicted using the snippets for KIC 6922244 . 
The code snippets work on Google colab.

Could it some kind of incompatibility issue between modules? E.g., recent lightkurve only works with astropy 5.0+

You could help to narrow down the problem by reporting the environment you use.

Environment
```python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import lightkurve as lk; print("lightkurve", lk.__version__)
import astropy; print("astropy", astropy.__version__)
import numpy as np; print("numpy", np.__version__)
```

Example output (on google colab):
(requires `!pip install lightkurve` to run lightkurve there.)

Linux-5.10.133+-x86_64-with-glibc2.27
Python 3.8.16 (default, Dec  7 2022, 01:12:13) 
[GCC 7.5.0]
lightkurve 2.3.0
astropy 5.1.1
numpy 1.21.6

**Comment by cdevaneprugh** [comment]
Linux-5.15.0-56-generic-x86_64-with-glibc2.35
Python 3.9.13 (main, Aug 25 2022, 23:26:10) 
[GCC 11.2.0]
lightkurve 2.3.0
astropy 5.1
numpy 1.21.5

Also, for what its worth I'm using conda version 22.9.0

**Comment by orionlee** [comment]
Hmm, I am stumped. Your environment is quite similar to the google colab, and the codes worked there.

To be doubly sure, I have put the code snippet (with working output) in the following colab nottebook:

https://colab.research.google.com/drive/1mIJdfAUTek_PJCMUPpcXSQOL2QUTDhJi?usp=sharing

Could you try it out and ensure the snippet is reproducing what you intended?

     On Thursday, December 8, 2022, 03:21:06 PM PST, Cooper DeVane-Prugh ***@***.***> wrote:  
 
 


Linux-5.15.0-56-generic-x86_64-with-glibc2.35
Python 3.9.13 (main, Aug 25 2022, 23:26:10)
[GCC 11.2.0]
lightkurve 2.3.0
astropy 5.1
numpy 1.21.5

Also, for what its worth I'm using conda version 22.9.0

**Comment by cdevaneprugh** [comment]
The code functions fine in the colab notebook you sent. It brakes in my jupyter notebooks or if I run a script through the terminal.

**Comment by cdevaneprugh** [comment]
Is it possibly an issue from anaconda handling all my python packages and dependencies? I can try it in a VM using pip to install everything and see what happens.

**Comment by orionlee** [comment]
> Is it possibly an issue from anaconda handling all my python packages and dependencies? I can try it in a VM using pip to install everything and see what happens.

I'd suggest you tried different environments to see what happened. Possibilities I could think of:
- a new (minimal for lightkurve) conda env
- using pip (or pipenv, etc.)
- VM + pip (or VM + conda)

In addition to google colab, I also tried another 2 environments, and still could not reproduce the problem.
1. Windows 11 + Conda (with conda-forge)
2. Windows WSL2 + pipenv


#### Windows 11 + Conda (with conda-forge)

Windows-10-10.0.22621-SP0
Python 3.9.10 | packaged by conda-forge | (main, Feb  1 2022, 21:21:54) [MSC v.1929 64 bit (AMD64)]
lightkurve 2.4.0dev
astropy 5.0.4
numpy 1.22.3


#### Windows WSL2 + pipenv
Linux-5.15.79.1-microsoft-standard-WSL2-x86_64-with-glibc2.29
Python 3.8.10 (default, Jun 22 2022, 20:18:18)
[GCC 9.4.0]
lightkurve 2.1.0
astropy 5.0.2
numpy 1.22.3

**Comment by Nschanche** [comment]
Hi, I still have the issue. Here is the environment data:

macOS-12.6-arm64-arm-64bit
Python 3.9.13 (main, Aug 25 2022, 18:24:45) 
[Clang 12.0.0 ]
lightkurve 2.3.0
astropy 5.2.1
numpy 1.24.1

**Comment by orionlee** [comment]
Try the following workaround: downgrade numpy to 1.23.5 , it might help to resolve the issue.

Details:
- I created a brand new environment on Windows (with numpy 1.24.1, and hit an error in `remove_outliers()` [*]
- when I downgrade numpy to 1.23.5, there is no problem.

[*] The error I got is a different error (`TypeError: cannot write to unmasked output`), so there is some chance the downgrade might not solve the problem you faced.


Versions that worked.
```
Windows-10-10.0.22621-SP0
Python 3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 08:41:22) [MSC v.1929 64 bit (AMD64)]
Numpy 1.23.5
pyerfa 2.0.0.1
astropy 5.2.1
Scipy 1.10.0
Matplotlib 3.6.3
lightkurve 2.3.0
```

Test codes:
```python
lc = lk.search_lightcurve('KIC 6922244', author="Kepler", 
                          cadence="long", quarter=4).download(download_dir="data")
print(len(lc))
lc = lc.remove_outliers()
print(len(lc))
```

**Comment by orionlee** [comment]
@Nschanche if you still have the issue, could you also run the following codes to see if it reproduces the error you countered?


```python
import numpy as np
import astropy
from astropy.stats.sigma_clipping import sigma_clip
from astropy import units as u
from astropy.utils.masked import Masked

ma = Masked([1., 2., 3., np.nan], mask=[False, False, True, True])
ma = ma * u.m  # convert to MaskedQuantity
display(ma)

res = sigma_clip(ma)
res
```

**Comment by orionlee** [comment]
PR #1279, #1280 should have a fix, and is released in lightkurve v2.4.0. So I'm closing the issue.
